core/src/main/java/by/ilyin/core/util/validation/custom/ChangeEmailParametersValidator.java

```diff
@@ -0,0 +1,59 @@
+package by.ilyin.core.util.validation.custom;
+
+import by.ilyin.core.dto.ChangeEmailDTO;
+import by.ilyin.core.entity.CustomUser;
+import by.ilyin.core.exception.http.CustomConstraintValidationException;
+import by.ilyin.core.repository.CustomUserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.constraintvalidation.SupportedValidationTarget;
+import javax.validation.constraintvalidation.ValidationTarget;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@SupportedValidationTarget(ValidationTarget.PARAMETERS)
+public class ChangeEmailParametersValidator implements ConstraintValidator<ConsistentChangeEmailParameters, Object[]> {
+
+    private final CustomUserRepository customUserRepository;
+
+    @Override
+    public boolean isValid(Object[] value, ConstraintValidatorContext context) {
+        boolean isConsistentParameters = Boolean.TRUE;
+        List<String> errorMessage = new ArrayList<>();
+        HttpStatus httpStatus = null;
+        if (value.length >= 2 &&
+                value[0] instanceof Long &&
+                value[1] instanceof ChangeEmailDTO) {
+            Long id = (Long) value[0];
+            ChangeEmailDTO changeEmailDTO = (ChangeEmailDTO) value[1];
+            Optional<CustomUser> optionalUserById = customUserRepository.findById(id);
+            if (optionalUserById.isEmpty()) {
+                httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
```

Comment:
@YauhenIlyin for what reason did you return especially INTERNAL_SERVER_ERROR?

Reply:
@DzmitryKazakevich 
because we get the id from the token, the absence of this user in the database is our internal error

Reply:
if the server is running correctly, the user will always be found